doc/tutorials/understanding-error-messages.md

```diff
@@ -46,3 +46,59 @@ Every category comes with its own solution approach which is intended to be as g
    Validation errors are heavily used for Elektra's `configuration specification`
    feature and should tell users that their given input does not match a certain
    pattern/type/expected semantic. Retry but with a conforming configuration value.
+
+## Error message format
+
+Error messages which are emitted from Elektra's C/C++ code have the same message format for all errors
```

Comment:
Why should this be specific for C/C++?

Reply:
How do you say "everything non-tool related"? What I have seen is that only tools may yield other error messages

Reply:
Tools yielding other error messages are simply a bug that have not yet been fixed.